modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/ProductOptionDTOConverter.java

```diff
@@ -45,13 +46,16 @@ public ProductOption toDTO(DTOConverterContext dtoConverterContext)
 			_cpDefinitionOptionRelLocalService.getCPDefinitionOptionRel(
 				(Long)dtoConverterContext.getId());
 
+		CPOption cpOption = cpDefinitionOptionRel.getCPOption();
+
 		String languageId = _language.getLanguageId(
 			dtoConverterContext.getLocale());
 
 		return new ProductOption() {
 			{
 				setDescription(
 					() -> cpDefinitionOptionRel.getDescription(languageId));
+				setExternalReferenceCode(cpOption::getExternalReferenceCode);
```

Comment:
because the CPOption is used only here is better to 
```
setExternalReferenceCode(
					() -> {CPOption cpOption = cpDefinitionOptionRel.getCPOption();; return cpOption.getExternalReferenceCode()});```